pkg/services/sqlstore/user_test.go

```diff
@@ -541,6 +541,28 @@ func TestUserDataAccess(t *testing.T) {
 					}
 				})
 			})
+
+			Convey("When creating a new user with an already existing email returns error", func() {
+				createUserCmd := &models.CreateUserCommand{
+					Email:        "user2@test.com",
+					Name:         "user2",
+					Login:        "user2",
+					SkipOrgSetup: true,
+				}
+				err := CreateUser(context.Background(), createUserCmd)
+				So(err, ShouldEqual, models.ErrUserAlreadyExists)
+			})
+
+			Convey("When creating a new user with an already existing login returns error", func() {
+				createUserCmd := &models.CreateUserCommand{
+					Email:        "user-2@test.com",
+					Name:         "user2",
```

Comment:
Where is user2 already inserted into the database here? Couldn't find it.

Reply:
Ditto 👉 https://github.com/grafana/grafana/pull/26958#discussion_r469511437